gobblin-core/src/main/java/gobblin/writer/AvroHdfsTimePartitionedWriter.java

```diff
@@ -172,9 +170,11 @@ public void write(GenericRecord record) throws IOException {
     // If the path is in pathToWriterMap simply retrieve the writer, and write the record
     if (!this.pathToWriterMap.containsKey(writerOutputPath)) {
 
-      LOG.info("Creating a new DataWriter for path: " + new Path(this.baseFilePath, writerOutputPath));
+      Path writerOutputPathWithBase = new Path(this.baseFilePath, writerOutputPath);
+
+      LOG.info("Creating a new DataWriter for path: " + writerOutputPathWithBase);
```

Comment:
Previously this log output was actually incorrect (outputted e.g. CaptchaEvent/CaptchaEvent/hourly/2015/06/29/01 - the baseFilePath was doubled). Having the writerOutputPath contain the baseFilePath also created an inconsistency since WriterUtils.getWriterOutputDir already appends the baseFilePath onto the end of the output directory so I refactored this a little. This is why getPathForColumnValue changed below (the javadoc comment for getPathForColumnValue didn't seem to imply that it would be adding the baseFilePath, anyway)